hack/test-cmd.sh

```diff
@@ -401,6 +401,21 @@ runTests() {
   # Post-condition: valid-pod POD has image kubernetes/pause
   kube::test::get_object_assert pods "{{range.items}}{{$image_field}}:{{end}}" 'kubernetes/pause:'
 
+  ## If resourceVersion is specified in the patch, it will be treated as a precondition, i.e., if the resourceVersion is different from that is stored in the server, the Patch should be rejected
+  ERROR_FILE="${KUBE_TEMP}/conflict-error"
+  resourceVersion=$(kubectl get "${kube_flags[@]}" pod valid-pod -o go-template='{{ .metadata.resourceVersion }}')
+  ((resourceVersion+=10))
+  # Command
+  kubectl patch "${kube_flags[@]}" pod valid-pod -p='{"spec":{"containers":[{"name": "kubernetes-serve-hostname", "image": "nginx"}]},"metadata":{"resourceVersion":"'$resourceVersion'"}}' 2> "${ERROR_FILE}" || true
```

Comment:
We can just put `! kubectl patch...` to imply that this command should fail (and then we don't need L410-417). 
Is there a test for the "resourceVersion is NOT different from the one stored in the server" case?